base/strings/search.jl

```diff
@@ -194,7 +217,7 @@ function _searchindex(s::ByteArray, t::ByteArray, i::Integer)
     bloom_mask = UInt64(0)
     skip = n - 1
     tlast = _nthbyte(t,n)
```

Comment:
The `_nthbyte(t,index)` calls can just be `t[index]` — the `_nthbyte` function is only needed if `t` might be a string instead of an array.

Reply:
Alternatively, we could define:
```
_nthbyte(t::AbstractVector,index) = t[index + (1 - firstindex(t))]
```
to allow us to use 1-based indexing (get rid of `firstindex` below and disregard my comment about `lastindex`).  This might be simpler, and there should be no runtime cost for `Vector` since the compiler should be able to eliminate the ` + (1 - firstindex(t))` operation.